### DISCUSSION TITLE
Cloning APK

### DISCUSSION THREAD

**Discussion by kefir500** [kefir500/apk-editor-studio#39] [Ideas]
## Introduction

APK Editor Studio v1.5.0 release is finally out. It includes the long-awaited ability to clone APK (or, more technically, change the package name). However, the current state of this feature is marked experimental: certain applications are still not supported since the lack of information on this topic leaves us with no stable instructions or algorithms.

In this discussion, I would like to collect the community's knowledge that will help to improve the app cloning feature in APK Editor Studio or even help other users looking for the corresponding information for their own purposes.

## Disclaimer

Neither the APK cloning feature nor this particular discussion encourages piracy. When cloning an Android application, make sure that its license allows this procedure.

## Guides and Instructions

- [XDA Developers](https://forum.xda-developers.com/t/how-to-guide-mod-change-package-names-of-apks.2760965/) – popular on the web (but outdated) guide
- [Notes by iBotPeaches](https://github.com/iBotPeaches/Apktool/issues/2538) – additional notes from the current Apktool developer

#### `AndroidManifest.xml`
- All `old.package.name` entries are replaced with `new.package.name`.

#### `res/`
- All `old.package.name` entries are replaced with `new.package.name`.

#### `smali*/`
- All `old.package.name` entries are replaced with `new.package.name`.
- All `Lold/package/name` entries are replaced with `Lnew/package/name`.
- `old/package/name` directory is renamed to `new/package/name`.
- These changes are applied to `smali` and all `smali_classesN` directories.

#### `unknown/`
- Files in this directory are not modified.

## Share your Experience

**APK Editor Studio is an open-source project, and it's within the power of the community to make it better.**

Any help, tips, personal experience, and observations are welcome. Feel free to join the discussion.

**Comment by StifflersMoM69** [comment]
Posted an issue here https://github.com/kefir500/apk-editor-studio/issues/128

**Comment by jrb80** [comment]
I am getting this error whilst trying to clone. It seems to get most of the way through processing the clone but bombs out mid way through. It has worked previously on other APKs. I have tried different Windows clients and therefore suggest it could be a APK specific issue.

Any suggestions to resolve? Is there a log file somewhere that can point to why the clone is failing?

**Reply by jrb80** [reply]
@kefir500 fyi, I passed the same APK through https://clonemy.app/ and it worked fine. So I am guessing there is something amiss with APK Editor Studio.

**Reply by kefir500** [reply]
Unfortunately, there is no source code for this tool. However, according its description, it is using nearly the same method as APK Editor Studio. Actually, it would be helpful to compare both APKs generated by each tool. And if APK Editor Studio doesn't generate the output APK, it would also be helpful to attach this APK so I could investigate it.

**Comment by Ibuprophen** [comment]
I was curious about the CloneMy.App and encountered the Developer's XDA Support thread [[TOOL] CloneMyApp - Android APK Cloner](https://xdaforums.com/t/tool-clonemyapp-android-apk-cloner.4513777/). 

Just wanted to let you know my friend @kefir500.. 

~Ibuprophen

**Comment by anlexN** [comment]
I use **Tools > Clone APK** in the **apk editor studio** to change package name `com.vkontakte.android` to `kom.vkontakte.android`.

<img width="1920" height="1020" alt="1" src="https://github.com/user-attachments/assets/d01decf8-bfb1-491a-92f1-9a65b06f1d75" />
<img width="1920" height="1020" alt="2" src="https://github.com/user-attachments/assets/92b76893-1fb8-420b-92f5-20efe8160d39" />
<img width="1920" height="1020" alt="3" src="https://github.com/user-attachments/assets/b71b50d3-af53-4e8c-b471-6aa1e42e8827" />
<img width="1920" height="1020" alt="4" src="https://github.com/user-attachments/assets/e6256429-b1fd-4479-aca3-7387f1cdd3a0" />


But I still can not install this modified app, how to solve these error:

```console
Serving...
All files should be loaded. Notifying the device.
adb: failed to install D:\thunder\kom.vkontakte.android-vk-8-144 - Copy.apk: Failure [INSTALL_FAILED_DUPLICATE_PERMISSION: Package kom.vkontakte.android attempting to redeclare permission com.vk.blockstore.permission.DELETE already owned by com.vkontakte.android]
Performing Incremental Install
Failure [INSTALL_FAILED_DUPLICATE_PERMISSION: Package kom.vkontakte.android attempting to redeclare permission com.vk.blockstore.permission.DELETE already owned by com.vkontakte.android]
Performing Streamed Install
```

**Comment by anlexN** [comment]
@jrb80 @kefir500 @lbuprophen 

Hello everyone, Can you help me? please
